homeassistant/components/xiaomi_tv/media_player.py

```diff
@@ -14,50 +15,56 @@
 import homeassistant.helpers.config_validation as cv
 
 DEFAULT_NAME = "Xiaomi TV"
+CONF_ASSUME_STATE = "assume_state"
 
 _LOGGER = logging.getLogger(__name__)
 
-SUPPORT_XIAOMI_TV = SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+SUPPORT_XIAOMI_TV = (
+    SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+)
 
 # No host is needed for configuration, however it can be set.
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_ASSUME_STATE, default=True): cv.boolean,
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Xiaomi TV platform."""
 
     # If a hostname is set. Discovery is skipped.
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME)
+    assume_state = config.get(CONF_ASSUME_STATE)
 
     if host is not None:
-        # Check if there's a valid TV at the IP address.
-        if not pymitv.Discover().check_ip(host):
-            _LOGGER.error("Could not find Xiaomi TV with specified IP: %s", host)
-        else:
-            # Register TV with Home Assistant.
-            add_entities([XiaomiTV(host, name)])
+        # Register TV with Home Assistant.
+        async_add_entities([XiaomiTV(host, name, assume_state)])
     else:
         # Otherwise, discover TVs on network.
-        add_entities(XiaomiTV(tv, DEFAULT_NAME) for tv in pymitv.Discover().scan())
+        async_add_entities(
+            XiaomiTV(tv, DEFAULT_NAME, assume_state) for tv in pymitv.Discover().scan()
+        )
 
 
 class XiaomiTV(MediaPlayerDevice):
     """Represent the Xiaomi TV for Home Assistant."""
 
-    def __init__(self, ip, name):
+    def __init__(self, ip, name, assume_state):
         """Receive IP address and name to construct class."""
 
         # Initialize the Xiaomi TV.
-        self._tv = pymitv.TV(ip)
+        self._tv = pymitv.TV(ip, assume_state=assume_state)
```

Comment:
Please instantiate the TV in `setup_platform` and pass the TV to the entity when creating the entity.